### PR TITLE
Add support for listing source_transactions

### PR DESCRIPTION
**Pull request by ob-stripe**
r? @brandur-stripe
cc @stripe/api-libraries @stan-stripe

Adds support for the `/v1/sources/src_.../source_transactions` endpoint.

`source_transaction` objects, like `source` objects, have a `type` attribute and an attribute named with the value of `type` and containing another map. This was previously handled in `SourceDeserializer`. In order to avoid duplicating that code, I turned it into a generic `TypeDataDeserializer` used by both the `Source` and `SourceTransaction` classes.

### CODE REVIEW

File: src/main/java/com/stripe/model/TypeDataDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package com.stripe.model;
+
+import com.google.gson.*;
+
+import java.lang.reflect.Type;
+import java.util.Map;
+import java.util.HashMap;
+
+public class TypeDataDeserializer<T extends HasTypeData> implements JsonDeserializer<T> {
```
Comment (brandur-stripe) [inline src/main/java/com/stripe/model/TypeDataDeserializer.java]: Would you mind giving this a slightly more specific name and adding a doc block explaining roughly what it does to the class? I think that "type data deserializer" is just a little too generic given that this is all related to sources.

Sorry definitely a nit given that you were not the one to originally cause these issues :$

File: src/main/java/com/stripe/model/HasTypeData.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.stripe.model;
+
+import java.util.Map;
+
+public interface HasTypeData {
```
Comment (brandur-stripe) [inline src/main/java/com/stripe/model/HasTypeData.java]: Good call on the interface!

I wonder if we should call this `HasSourceTypeData` or something too? (Again, will likely always be sources specific.)

**Comment by brandur-stripe** [comment]
Left a couple comments, but great work as always!

**Comment by ob-stripe** [comment]
I renamed the various `TypeData` references to `SourceTypeData`, and left a comment on the deserializer class as request. ptal @brandur-stripe

**Comment by brandur-stripe** [comment]
Thanks Olivier! Shipping.

**Comment by brandur-stripe** [comment]
Released as 5.22.0.

**Pull request merged**